elasticsearch_dsl/document.py

```diff
@@ -15,474 +15,6 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
-import collections.abc
-from fnmatch import fnmatch
+from elasticsearch_dsl._sync.document import *  # noqa: F401, F403
```

Comment:
Can you please check if common tools such as mypy, pyright and jedi are able to autocomplete/type check correctly despite the star import?

Reply:
That is a good point. I'll check, but I suspect we'll have to do an explicit import here.

Reply:
I have added explicit imports to document.py, search.py, update_by_query.py and mapping.py.

Question: Do we want to also import the async classes on these files? Because right now the async imports are all long and awkward:

```python
from elasticsearch_dsl._async.search import AsyncSearch
```

We could change this to:

```python
from elasticsearch_dsl.search import AsyncSearch
```

But of course, if we do this, we would always be import the sync and async versions into the Python environment, regardless of which one you want to use. Given that all the common methods are in the base class, this shouldn't add a lot of overhead, I think.

Reply:
As hinted above in our discussion last week, I think what should be encouraged is:

```python
from elasticsearch_dsl import Search
from elasticsearch_dsl import AsyncSearch
```

We keep `search` only because it used to work but I would not encourage it:

```python
from elasticsearch_dsl.search import Search
```

And hopefully nobody will use this as `_async` starts as an underscore to show that it's private:

```python
from elasticsearch_dsl._async.search import AsyncSearch
```